### PR TITLE
tests: loopback & gpio: add edge-case and boundary validation

### PR DESCRIPTION
**Pull request by CPPavithra**
Hi @Ayush1325 ! I've been working on expanding the integration tests for Greybus on native_sim.

This PR adds several edge-case checks:
        1. Loopback: Added zero-length transfers, invalid operation IDs (0x7F), and a simple rapid-ping test.
        2. GPIO: Added a boundary check for invalid pin indices.

 NOTE:
While I was writing these tests, I started digging into the driver code to see how the messages were being handled. I noticed a discrepancy in `subsys/greybus/gpio.c:86 `within the `gb_gpio_get_direction` function (and likely other handlers like `gb_gpio_set_value`).

The current implementation takes the pin index (request->which) from the Greybus message and passes it directly to the Zephyr GPIO API:
`resp_data.direction = gpio_pin_is_input(dev, request->which);`

Since there’s no bounds check against `data->ngpios`, my test for Pin 255 returns `SUCCESS` instead of an error, even though that pin doesn't exist on the simulated hardware. I've included the failing test for now to highlight this.

If you agree this should be validated at the Greybus layer, I’m happy to send a follow-up PR to add those checks across the GPIO service.

Verified with `west build` on `native_sim`.

### CODE REVIEW

File: tests/greybus/integration/gpio/src/main.c [hunk 1]
```diff
@@ -181,3 +181,26 @@ ZTEST(greybus_gpio_tests, test_set_value)
...
+    struct gb_message *msg= gb_message_request_alloc(sizeof(*req_data), GB_GPIO_TYPE_GET_DIRECTION, false);
+
+    req_data= (struct gb_gpio_get_direction_request *)msg->payload;
+    req_data->which= 255;
+
+    greybus_rx_handler(1, gb_message_copy(msg));
+    resp= gb_transport_get_message();
+
+    //a failure is expected
+    bool success= gb_message_is_success(resp.msg);
```
Comment (Ayush1325) [inline tests/greybus/integration/gpio/src/main.c]: No need to create intermediate variable here. Directly call it in `zassert_false`. It is fine to leak the message in case of failing since this is just a test.

**Review by Ayush1325 — COMMENTED** [review]
Can you run the changes through clang-format. You can use the following command for just formatting the lines changed in the last commit:

```
git diff -U0 --no-color --relative HEAD^ | clang-format-diff -p1 -i
```

File: tests/greybus/integration/loopback/src/main.c [hunk 2]
```diff
@@ -92,3 +92,62 @@ ZTEST(greybus_loopback_tests, test_transfer)
...
+{
+    struct gb_msg_with_cport resp;
+    struct gb_message *req= gb_message_request_alloc(sizeof(struct gb_loopback_transfer_request),GB_LOOPBACK_TYPE_TRANSFER, false);
+    struct gb_loopback_transfer_request *req_data= (struct gb_loopback_transfer_request *)req->payload;
+
+    req_data->len = sys_cpu_to_le32(0);
+    //first we send an empty request
+    greybus_rx_handler(1, req);
+    resp = gb_transport_get_message(); //receiving
+
```
Comment (Ayush1325) [inline tests/greybus/integration/loopback/src/main.c]: Remove extra newlines

**Comment by CPPavithra** [comment]
> > > Can you run the changes through clang-format. You can use the following command for just formatting the lines changed in the last commit:
> > > ```
> > > git diff -U0 --no-color --relative HEAD^ | clang-format-diff -p1 -i
> > > ```
> > 
> > 
> > I have implemented the bounds check for `gb_gpio_get_direction` and updated the function signature to pass the data pointer for consistency. If this looks good to you, I can quickly roll it out to the other GPIO handlers as well! I also cleaned up the test code as suggested and ran everything through clang-format.
> > All integration tests are now passing successfully on `native_sim`.
> 
> Sure, feel free to add the check to other places in gpio.

I have expanded the boundary checks across all GPIO handlers, cleaned up the test variables and newlines, and applied clang-format and pushed the updated one.
